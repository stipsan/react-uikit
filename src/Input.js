import cx from 'classnames'
import { PropTypes } from 'react'

const Input = ({
  blank,
  className: customClassName,
  danger,
  flip,
  icon,
  large,
  small,
  success,
  width,
  ...other
}) => {
  const className = cx(customClassName, {
    'uk-form-blank': blank,
    'uk-form-danger': danger,
    'uk-form-large': large,
    'uk-form-small': small,
    'uk-form-success': success,
    'uk-form-width-large': width === 'large',
    'uk-form-width-medium': width === 'medium',
    'uk-form-width-mini': width === 'mini',
    'uk-form-width-small': width === 'small',
    'uk-width-1-1': width === 'full',
  })
  const input = <input {...other} className={className} />
  if (icon) {
    return (
      <div className={cx('uk-form-icon', { 'uk-form-icon-flip': flip })}>
        <i
          className={cx(`uk-icon-${icon}`, {
            'uk-icon-spin': icon === 'spinner' || icon === 'refresh',
          })}
        />
        {input}
      </div>
    )
  }
  return input
}

Input.propTypes = {
  blank: PropTypes.bool,
  className: PropTypes.string,
  danger: PropTypes.bool,
  flip: PropTypes.bool,
  icon: PropTypes.string,
  large: PropTypes.bool,
  small: PropTypes.bool,
  success: PropTypes.bool,
  width: PropTypes.oneOf(['full', 'large', 'medium', 'small', 'mini', false]),
}

Input.defaultProps = {
  blank: false,
  className: '',
  danger: false,
  flip: false,
  icon: '',
  large: false,
  small: false,
  success: false,
  width: false,
}

export default Input
