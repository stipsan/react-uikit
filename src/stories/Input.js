import { storiesOf } from '@kadira/storybook'
import { Input } from 'uikit-react'

// This is to work around: https://github.com/kadirahq/react-storybook-addon-info/issues/26#issuecomment-229029177
Input.displayName = 'Input'

storiesOf('Input', module)
  .addWithInfo('Basic Usage', `
  ---
  For more advanced use cases please check out [\`redux-form-uikit\`](http://form.uikit-react.io).
  `, () => (
    <form className="uk-margin-bottom uk-form">
      <Input placeholder="Text input" />
      &nbsp;
      <Input danger defaultValue="danger" />
      &nbsp;
      <Input success defaultValue="success" />
      &nbsp;
      <Input blank placeholder="blank" />
    </form>
  ), { header: false, inline: true, propTables: [Input], mtrcConf: { // eslint-disable-line
    a: ({ href, children }) => <a href={href} target="_blank">{children}</a>, // eslint-disable-line
  } })
  .addWithInfo('Size modifiers', '', () => (
    <form className="uk-margin-bottom uk-form">
      <div className="uk-margin">
        <Input large placeholder="large" />
      </div>
      <div className="uk-margin">
        <Input placeholder="default" />
      </div>
      <div className="uk-margin">
        <Input small placeholder="small" />
      </div>
    </form>
  ), { header: false, inline: true })
  .addWithInfo('Width modifiers', '', () => (
    <form className="uk-margin-bottom uk-form">
      <div className="uk-margin">
        <Input placeholder="full" width="full" />
      </div>
      <div className="uk-margin">
        <Input placeholder="large" width="large" />
      </div>
      <div className="uk-margin">
        <Input placeholder="medium" width="medium" />
      </div>
      <div className="uk-margin">
        <Input placeholder="small" width="small" />
      </div>
    </form>
  ), { header: false, inline: true })
  .addWithInfo('Form and icons', '', () => (
    <form>
      <div className="uk-form-row">
        <div className="uk-margin">
          <Input icon="calendar" />
          &nbsp;
        </div>
        <div className="uk-margin">
          <Input flip icon="clock" />
        </div>
      </div>
      <div className="uk-form-row">
        <div className="uk-margin">
          <Input icon="twitter" />
          &nbsp;
        </div>
        <div className="uk-margin">
          <Input flip icon="spinner" />
        </div>
      </div>
    </form>
  ), { header: false, inline: true })
