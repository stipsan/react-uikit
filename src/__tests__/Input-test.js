import renderer from 'react-test-renderer'

import Input from '../Input'

describe('Input', () => {
  it('renders correctly', () => {
    expect(renderer.create(
      <Input />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input className="uk-search-field" type="search" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input danger />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input success />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input blank />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input large />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input small />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input width="full" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input width="large" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input width="medium" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input width="small" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input icon="calendar" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input flip icon="calendar" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input icon="spinner" />
    ).toJSON()).toMatchSnapshot()

    expect(renderer.create(
      <Input flip icon="refresh" />
    ).toJSON()).toMatchSnapshot()
  })
})
