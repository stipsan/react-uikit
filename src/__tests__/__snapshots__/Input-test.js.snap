// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<input
  className="uk-input "
/>
`;

exports[`Input renders correctly 2`] = `
<input
  className="uk-input uk-search-field"
  type="search"
/>
`;

exports[`Input renders correctly 3`] = `
<input
  className="uk-input  uk-form-danger"
/>
`;

exports[`Input renders correctly 4`] = `
<input
  className="uk-input  uk-form-success"
/>
`;

exports[`Input renders correctly 5`] = `
<input
  className="uk-input  uk-form-blank"
/>
`;

exports[`Input renders correctly 6`] = `
<input
  className="uk-input  uk-form-large"
/>
`;

exports[`Input renders correctly 7`] = `
<input
  className="uk-input  uk-form-small"
/>
`;

exports[`Input renders correctly 8`] = `
<input
  className="uk-input  uk-width-1-1"
/>
`;

exports[`Input renders correctly 9`] = `
<input
  className="uk-input  uk-form-width-large"
/>
`;

exports[`Input renders correctly 10`] = `
<input
  className="uk-input  uk-form-width-medium"
/>
`;

exports[`Input renders correctly 11`] = `
<input
  className="uk-input  uk-form-width-small"
/>
`;

exports[`Input renders correctly 12`] = `
<div
  className="uk-inline"
>
  <span
    className="uk-icon uk-form-icon"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M 2,3 2,17 18,17 18,3 2,3 Z M 17,16 3,16 3,8 17,8 17,16 Z M 17,7 3,7 3,4 17,4 17,7 Z\\"></path><rect width=\\"1\\" height=\\"3\\" x=\\"6\\" y=\\"2\\"></rect><rect width=\\"1\\" height=\\"3\\" x=\\"13\\" y=\\"2\\"></rect>",
        }
      }
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    />
  </span>
  <input
    className="uk-input "
  />
</div>
`;

exports[`Input renders correctly 13`] = `
<div
  className="uk-inline"
>
  <span
    className="uk-icon uk-form-icon uk-form-icon-flip"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M 2,3 2,17 18,17 18,3 2,3 Z M 17,16 3,16 3,8 17,8 17,16 Z M 17,7 3,7 3,4 17,4 17,7 Z\\"></path><rect width=\\"1\\" height=\\"3\\" x=\\"6\\" y=\\"2\\"></rect><rect width=\\"1\\" height=\\"3\\" x=\\"13\\" y=\\"2\\"></rect>",
        }
      }
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    />
  </span>
  <input
    className="uk-input "
  />
</div>
`;

exports[`Input renders correctly 14`] = `
<div
  className="uk-inline"
>
  <span
    className="uk-icon uk-form-icon uk-form-icon-flip"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"1.1\\" d=\\"M17.08,11.15 C17.09,11.31 17.1,11.47 17.1,11.64 C17.1,15.53 13.94,18.69 10.05,18.69 C6.16,18.68 3,15.53 3,11.63 C3,7.74 6.16,4.58 10.05,4.58 C10.9,4.58 11.71,4.73 12.46,5\\"></path><polyline fill=\\"none\\" stroke=\\"#000\\" points=\\"9.9 2 12.79 4.89 9.79 7.9\\"></polyline>",
        }
      }
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    />
  </span>
  <input
    className="uk-input "
  />
</div>
`;
